lib/rubocop/cop/sorbet/signatures/signature_build_order.rb

```diff
@@ -12,6 +12,25 @@
 module RuboCop
   module Cop
     module Sorbet
+      # Checks for the correct order of sig builder methods:
+      # - abstract, override, or overridable
+      # - type_parameters
+      # - params
+      # - returns, or void
+      # - soft, checked, or on_failure
```

Comment:
`soft` has been renamed into `on_failure` [a while ago](https://github.com/sorbet/sorbet/pull/1141) but can theoretically still be found in codebases using an old Sorbet.

Reply:
> I did notice that there are a couple methods which we don't include in our list, which we should probably add:
> 
> * final
> * implementation
> * bind

Let's add them 👍

Reply:
Sounds good. I've opened #189 to do that, to unblock this PR (and the subsequent ones branched off of it, and keep it about the documentation, especially since I need to figure out what `implementation` is and where it belongs 😅).